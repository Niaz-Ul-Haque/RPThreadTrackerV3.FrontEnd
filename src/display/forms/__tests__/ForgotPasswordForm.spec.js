// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../../config/tests/helpers.unit';
import ForgotPasswordForm from '../forgot-password/ForgotPasswordForm';
// #endregion imports

const createTestProps = propOverrides => ({
	handleInputChange: jest.fn(),
	...propOverrides
});

describe('rendering', () => {
	it('should render valid snapshot', () => {
		const props = createTestProps();
		const jsx = (<ForgotPasswordForm {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
	it('should validate the email field', () => {
		const props = createTestProps();
		const jsx = (<ForgotPasswordForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'email-field');
		expect(field.props().validate.required).toHaveProperty('value', true);
		expect(field.props().validate.email).toHaveProperty('value', true);
	});
});

describe('behavior', () => {
	it('should handle input change for email', () => {
		const handleInputChange = jest.fn();
		const props = createTestProps({ handleInputChange });
		const jsx = (<ForgotPasswordForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'email-field');
		field.simulate('change');
		expect(handleInputChange).toHaveBeenCalledTimes(1);
	});
});
