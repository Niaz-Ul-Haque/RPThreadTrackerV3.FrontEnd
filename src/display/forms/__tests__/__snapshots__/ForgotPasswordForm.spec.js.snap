// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render valid snapshot 1`] = `
<div>
  <div
    data-spec="email-field"
  >
    <AvField
      name="Email"
      onChange={[MockFunction]}
      placeholder="Email"
      type="text"
      validate={
        Object {
          "email": Object {
            "errorMessage": "Please enter a valid email.",
            "value": true,
          },
          "required": Object {
            "errorMessage": "You must enter an email.",
            "value": true,
          },
        }
      }
    />
  </div>
</div>
`;
